lib/ansible/modules/cloud/amazon/glue_job.py

```diff
@@ -0,0 +1,377 @@
+        return True
+    if 'ExecutionProperty' in user_params and user_params['ExecutionProperty']['MaxConcurrentRuns'] != current_params['ExecutionProperty']['MaxConcurrentRuns']:
+        return True
+    if 'MaxRetries' in user_params and user_params['MaxRetries'] != current_params['MaxRetries']:
+        return True
+    if 'Timeout' in user_params and user_params['Timeout'] != current_params['Timeout']:
+        return True
+
+    return False
+
+
+def create_or_update_glue_job(connection, module, glue_job):
+    """
+    Create or update an AWS Glue job
+
+    :param connection: AWS boto3 glue connection
+    :param module: Ansible module
+    :param glue_job: a dict of AWS Glue job parameters or None
+    :return:
+    """
+
+    changed = False
+    params = dict()
+    params['Name'] = module.params.get("name")
+    params['Role'] = module.params.get("role")
+    if module.params.get("allocated_capacity") is not None:
+        params['AllocatedCapacity'] = module.params.get("allocated_capacity")
+    if module.params.get("command_script_location") is not None:
+        params['Command'] = {'Name': module.params.get("command_name"), 'ScriptLocation': module.params.get("command_script_location")}
+    if module.params.get("connections") is not None:
+        params['Connections'] = {'Connections': module.params.get("connections")}
+    if module.params.get("default_arguments") is not None:
+        params['DefaultArguments'] = module.params.get("default_arguments")
+    if module.params.get("description") is not None:
+        params['Description'] = module.params.get("description")
+    if module.params.get("max_concurrent_runs") is not None:
+        params['ExecutionProperty'] = {'MaxConcurrentRuns': module.params.get("max_concurrent_runs")}
+    if module.params.get("max_retries") is not None:
+        params['MaxRetries'] = module.params.get("max_retries")
+    if module.params.get("timeout") is not None:
+        params['Timeout'] = module.params.get("timeout")
+
+    # If glue_job is not None then check if it needs to be modified, else create it
+    if glue_job:
+        if _compare_glue_job_params(params, glue_job):
+            try:
+                # Update job needs slightly modified params
+                update_params = {'JobName': params['Name'], 'JobUpdate': copy.deepcopy(params)}
+                del update_params['JobUpdate']['Name']
+                connection.update_job(**update_params)
+                changed = True
+            except (BotoCoreError, ClientError) as e:
+                module.fail_json_aws(e)
+    else:
+        try:
+            connection.create_job(**params)
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+    # If changed, get the Glue job again
+    if changed:
+        glue_job = _get_glue_job(connection, module, params['Name'])
+
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(glue_job))
```

Comment:
We've been having issues in other tests (`ec2_vpc_*`, `ec2_group`, etc) caused by AWS API's not being read-back consistent. I see the `_get_glue_job` call will silently return `None` if the job isn't found. Since at :312 and here you know the job exists, can you add a check that will retry if the job isn't found? I think camel2snake raises an exception if it's given a `None` as the parameter.

Reply:
Is there a specific way you'd like to implement that or just an `while glue_job is None` loop will do?

Reply:
Yeah, a simple loop like that would be perfect - I just want to protect against accidentally getting `None` from `_get_glue_job` because the AWS API hasn't seen the change yet.